openingd/dualopend.c

```diff
@@ -2704,53 +2746,12 @@ static u8 *opener_commits(struct state *state,
 					     /* Opener is local */
 					     LOCAL);
 
-	remote_commit = initial_channel_tx(state->channel, &wscript,
-					   state->channel,
-					   &state->first_per_commitment_point[REMOTE],
-					   REMOTE, direct_outputs, &error);
-
-	if (!remote_commit) {
-		*err_reason = tal_fmt(tmpctx, "Could not meet their fees"
-				      " and reserve: %s", error);
+	commit_msg = msg_for_remote_commit(state, state, &pbase, &error);
+	if (!commit_msg) {
+		*err_reason = tal_fmt(tmpctx, "%s", error);
```

Comment:
If you use `tmpctx` as the ctx arg to msg_from_remote_commit, you can just hand `err_reason` instead of `error` right?

Reply:
The error message inside of `remote_commit`/`initial_commit_tx` are all static strings, not allocated ones. So we're really sort of "type swapping" here from a static string to an alloc'd one.